src/caikit_nlp_client/http_client.py

```diff
@@ -292,3 +291,180 @@ def _create_json_request(self, model_id, text, **kwargs) -> dict[str, Any]:
+            },
+            "parameters": parameters,
+            "model_id": model_id,
+        }
+
+        req_kwargs = self._get_tls_configuration()
+
+        response = requests.post(
+            f"{self._api_base}/api/v1/task/sentence-similarity-tasks",
+            json=json_input,
+            timeout=timeout,
+            **req_kwargs,  # type: ignore
+        )
+        log.debug(f"Response: {response}")
+        return self._unpack_or_raise_details(response, details_field="detail")
+
+    def rerank(
+        self, model_id: str, documents: list[dict[str, Any]], query: str, timeout: float = 60.0, parameters: dict[str, Any]={}
+    ) -> dict[str, Any]:
+
+        if not model_id:
+            raise ValueError("request must have a model id")
+
+        log.info(f"Calling rerank for '{model_id}'")
+        json_input = {
+            "inputs": {
+                "documents": documents,
+                "query": query,
+            },
+            "parameters": parameters,
+            "model_id": model_id,
+        }
+
+        req_kwargs = self._get_tls_configuration()
+
+        response = requests.post(
+            f"{self._api_base}/api/v1/task/rerank",
+            json=json_input,
+            timeout=timeout,
+            **req_kwargs,  # type: ignore
+        )
+        log.debug(f"Response: {response}")
+        return self._unpack_or_raise_details(response, details_field="detail")
+
+    def rerank_tasks(
+        self, model_id: str, documents: list[dict[str, Any]], queries: list[str], timeout: float = 60.0, parameters: dict[str, Any]={}
+    ) -> dict[str, Any]:
+
+        if not model_id:
+            raise ValueError("request must have a model id")
+
+        log.info(f"Calling rerank_tasks for '{model_id}'")
+        json_input = {
+            "inputs": {
+                "documents": documents,
+                "queries": queries,
+            },
+            "parameters": parameters,
+            "model_id": model_id,
+        }
+
+        req_kwargs = self._get_tls_configuration()
+
+        response = requests.post(
+            f"{self._api_base}/api/v1/task/rerank-tasks",
+            json=json_input,
+            timeout=timeout,
+            **req_kwargs,  # type: ignore
+        )
+        log.debug(f"Response: {response}")
+        return self._unpack_or_raise_details(response)
+
+    def _unpack_or_raise_details(self, response: requests.Response, details_field="details") -> dict[str, Any]:
+        """
+        The details_field is present because there seems to be an inconsistent naming of this
+        field between different endpoints. Some have "details" others have "detail".
+        """
+        if response.status_code == 200:
+            return response.json()
+
+        try:
+            details = response.json()[details_field]
+        except json.JSONDecodeError:
+            details = f"{response.reason} {response.text=}"
+
+        raise RuntimeError(f"{response.status_code=} {details}")
```

Comment:
```suggestion
        try:
            details = response.json()[details_field]
        except json.JSONDecodeError as exc:
            details = f"{response.reason} {response.text=}"
            raise RuntimeError(f"{response.status_code=} {details}") from exc
```

Reply:
Not sure I understand? I tried to use the same pattern of error handling [already present](https://github.com/opendatahub-io/caikit-nlp-client/blob/c97f7d31f58d6feed60be71c09eb0d39c0eb98ef/src/caikit_nlp_client/http_client.py#L183) in the HTTP client

Reply:
Oh well that's fine, I guess we need to clean those up.